File: two_sum.c
/*
 * Note: The returned array must be malloced, assume caller calls free().
 */
int j = 0,*po;

int *create(int k, int j)
{
    int *ptr = (int *)malloc(2 * sizeof(int));
    ptr[0] = j;
    ptr[1] = k;
    return ptr;
}

int *twoSum(int *nums, int numsSize, int target, int *returnSize)
{
    int diff = target-nums[j],k = 0;
    for (int i = j + 1; i < numsSize; i++)
    {
        if (diff == nums[i])
        {
            po = create(i, j);
            *returnSize = 2;
            k = 1;
            break;
        }
    }
    if (k == 0)
    {
        j++;
        return twoSum(nums, numsSize, target, returnSize);
    }
    j=0;
    return po;
}

File: 1) Two Sum/two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int arr[] = new int[2];
        int diff,l = 0;
        for(int i = 0;i<nums.length-1;i++){
            diff = target-nums[i];
            for(int k = i+1;k<nums.length;k++){
                if(nums[k] == diff){
                    arr[0] = i;
                    arr[1] = k;
                    l = 1;
                    break;
                }
            }
            if(l==1)
                break;
        }
        return arr;
    }
}